pkg/host/state_machine.go

```diff
@@ -0,0 +1,133 @@
+package host
+
+import (
+	"context"
+	"sync/atomic"
+)
+
+// HostStatus is the status of the host
+type HostStatus string
+
+// Equals returns true if given SNRHostStatus matches this one
+func (s HostStatus) Equals(other HostStatus) bool {
+	return s == other
+}
+
+// IsNotIdle returns true if the SNRHostStatus != Status_IDLE
+func (s HostStatus) IsIdle() bool {
+	return s != Status_IDLE
```

Comment:
Function is called `IsIdle` but is returning "is not idle"

Reply:
Good catch